pTk/mTk/xlib/X11/X.h

```diff
@@ -179,13 +181,20 @@ are reserved in the protocol for errors and replies. */
 
 #define ShiftMask		(1<<0)
 #define LockMask		(1<<1)
-#define ControlMask		(1<<2)
+#ifndef _WIN32
+#   define ControlMask		(1<<2) /* See bug [9e31fd9449] and below */
+#endif
```

Comment:
This means that on Win32 `ControlMask` won't be defined as a macro. But `X.xs` checks for `#ifdef ControlMask`, so I think `perl -Mblib -wE "use Tk::X qw(ControlMask); say ControlMask()"` won't work and output 4 on Windows.

Possible workaround:
```c
#ifndef ControlMask
#define ControlMask ControlMask /* uses the enum below */
#endif
```

Reply:
Thanks for pointing this out, I failed to notice this earlier. It looks like the same issue might apply to `None`.

Why does X.xs check for whether `ControlMask`, `None`, and other macros are defined? I do not know for certain. But at the time X.xs was introduced (698f508f40f233bb34df06e913c3fafb5baca03b), Perl/Tk did not bundle X.h. Presumably it instead relied on X.h from the system, and so the defensive approach in X.xs may have been to deal with inconsistencies between different vendors’ X11 implementations. But that seems like a historical concern. Can X.xs just assume `ControlMask`/`None` are defined, as apparently done by usage elsewhere in Perl/Tk?